src/core/DY.Core.Label.Derived.fst

```diff
@@ -89,6 +46,107 @@ val right_flows_to_meet:
 let right_flows_to_meet tr l1 l2 =
   meet_eq tr (meet l1 l2) l1 l2
 
+(*** Equational theory for join ***)
+
+val join_associative:
+  l1:label -> l2:label -> l3:label ->
+  Lemma
+  (((l1 `join` l2) `join` l3) == (l1 `join` (l2 `join` l3)))
+let join_associative l1 l2 l3 =
+  intro_label_equal ((l1 `join` l2) `join` l3) (l1 `join` (l2 `join` l3)) (fun tr ->
+    join_flows_to_left tr (l1 `join` l2) l3;
+    join_eq tr l1 (l2 `join` l3) ((l1 `join` l2) `join` l3);
+    join_flows_to_right tr l1 (l2 `join` l3);
+    join_eq tr (l1 `join` l2) l3 (l1 `join` (l2 `join` l3))
+  )
+
+val join_commutes:
+  l1:label -> l2:label ->
+  Lemma
+  ((l1 `join` l2) == (l2 `join` l1))
+let join_commutes l1 l2 =
+  intro_label_equal (l1 `join` l2) (l2 `join` l1) (fun tr -> ())
+
+val join_label_public:
+  l:label ->
+  Lemma
+  ((l `join` public) == public)
+let join_label_public l =
+  intro_label_equal (l `join` public) public (fun tr -> ())
+
+val join_public_label:
+  l:label ->
+  Lemma
+  ((public `join` l) == public)
+let join_public_label l =
+  intro_label_equal (public `join` l) public (fun tr -> ())
```

Comment:
Why do we need both directions, if we also have `join_commutes`?

Reply:
can this be resolved with an SMTPat?
Maybe even better: can we combine both statements in one lemma? When having to call one of the lemmas, I don't want to think about, on which side the public is on. E.g., 
```
Lemma ((l `join` public) == public /\ public == (public `join` l))
```

Reply:
I wouldn't risk adding an SMTPat on commutativity, it's looks like the kind of things that would over-trigger.
However this is a good point, I will combine `join_public_label` and `join_label_public` in a single lemma (and same for `join_secret`, `meet_public` and `meet_secret`)!

Reply:
Done.